matchms/filtering/clean_compound_name.py

```diff
@@ -0,0 +1,63 @@
+import re
+from ..typing import SpectrumType
+
+
+def clean_compound_name(spectrum_in: SpectrumType) -> SpectrumType:
+    """Clean compound name.
+
+    A list of frequently seen name additions that do not belong to the compound
+    name will be removed."""
+
+    def remove_non_compound_name_parts(name):
+        """Clean "name string by removing known parts that don't belong there."""
+        name = name.strip()
+        # remove type NCGC00180417-03_C31H40O16_
+        name = re.split(r"[A-Z]{3,}[0-9]{8,}-[0-9]{2,}_[A-Z,0-9]{4,}_", name)[-1]
+        # remove type NCGC00160232-01! or MLS001142816-01!
+        name = re.split(r"[A-Z]{3,}[0-9]{8,}-[0-9]{2,3}\!", name)[-1]
+        # remove type Massbank:EA008813 option1|option2|option3
+        name = re.split(r"((Massbank:)|(MassbankEU:))[A-Z]{2}[0-9]{5,6}.*\|", name)[-1]
+        # remove type Massbank:EA008813 or MassbankEU:EA008813
+        name = re.split(r"((Massbank:)|(MassbankEU:))[A-Z]{2}[0-9]{5,6}", name)[-1]
+        # remove type HMDB:HMDB00943-1336
+        name = re.split(r"HMDB:HMDB[0-9]{4,}-[0-9]{1,}", name)[-1]
+        # remove type MoNA:662599
+        name = re.split(r"MoNA:[0-9]{5,}", name)[-1]
+        # ReSpect:PS013405 option1|option2|option3...
+        name = re.split(r"ReSpect:[A-Z]{2,}[0-9]{6}.*\|", name)[-1]
+        # ReSpect:PS013405 option1
+        name = re.split(r"[A-Z]{2,}[0-9]{6}( )", name)[-1]
+        # remove type 0072_2-Mercaptobenzothiaz
+        name = re.split(r"^[0-9]{4}_", name)[-1]
+        # remove type nameofcompound_CID20_170920 or Spiraeoside_HCD30_170919
+        name = re.split(r"_((HCD)|(CID))[0-9]{2}_[0-9]{5,6}$", name)[0]
+
+        # Remove further non compound-name parts
+        parts_remove = ["Spectral Match to",
```

Comment:
If you like you could group these lines (36-40) in a function and call it 'remove_futher_non_compount_parts' for example. I think it would be nice, although it is a bit of a fuzzy name. If you can't think of a better name that describes what you do in these lines, then maybe leave it like this, or remove the blank line above because it implies that these lines are somehow doing something conceptually different than the ones before line 36.

Reply:
I broke it down into two functions as you suggested. 
``remove_parts_by_regular_expression`` and ``remove_known_non_compound_parts``